EIPS/eip-7733.md

```diff
@@ -0,0 +1,60 @@
+---
+eip: 7733
+title: Deactivate EIP-158
+description: Deactivates EIP-158 in order to avoid conflicts in a context in which statelessness and EIP-7702 are active
+author: Guillaume Ballet (@gballet)
+discussions-to: https://ethereum-magicians.org/t/eip-7733-eip-158-deactivation/20445
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-07-02
+requires: 158, 7702
+---
+
+## Abstract
+
+Deactivate [EIP-158](./eip-158.md).
+
+## Motivation
+
+After [EIP-6780](./eip-6780.md), the `SELFDESTRUCT` instruction was neutered, so that contracts could no longer be deleted. Contract deletions can still occur, however, via EIP-158. While [EIP-7702](./eip-7702.md) this can not happen, this latter EIP introduces the possibility for an empty EoA to have state. This is causing some issues when interfacing with verkle, for which state deletion is not possible.
+
+EIP-158 was meant as a temporary measure to combat the "Shanghai attacks". Now that this is attack has been mitigated by other means, it can be deactivated.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+|Name|Value|
+|----|-----|
+|`FORK_TIME`|TBD|
+
+At `block.timestamp > FORK_TIME`, EIP-158 rules no longer apply.
```

Comment:
> At `block.timestamp > FORK_TIME`, EIP-158 rules no longer apply.

EIP-161 is 8 years old. Its behaviour has been assumed by EIP authors and implementers. You really ought to explicitly spell out what you actually want to do, rather than make some vague comments about deactivating it. I'm also fairly confident that you meant something different than what you wrote implies.

EIP-161 (see previous comment) has 4 effects:

a. When an account is `CREATE`d the nonce is incremented.
b. The new account gas cost is not charged if the account wouldn't be created because of rule c.
c. If an account would come into existence in an _empty_ state, it remains non-existent.
d. Under certain triggering conditions, empty accounts that are marked for deletion and deleted at the the end of the transaction.

As it stands, rules a, b and c are uncontroversial. Rule d is now a nullity because all empty accounts have been deleted and it is impossible to create new ones.

Your motivation for writing this EIP seems related to the edgecase that occurs if EIP-7702 code setting is used with an EOA that doesn't exist. EIP-7702 does not explain what happens in this case, which is a significant deficiency. If this edgecase results in the creation of empty accounts, it is a serious DOS vulnerability because an account can be created for only 2500 gas (other methods require at least 21000).

Personally I am opposed to a version of EIP-7702 that creates empty accounts. If it is decided that EIP-7702 should create empty accounts, then we can then discuss changing the rules about them. Until then this EIP doesn't change the spec in a meaningful way.

Reply:
> As it stands, rules a, b and c are uncontroversial. Rule d is now a nullity because all empty accounts have been deleted and it is impossible to create new ones.

eip 7702 changes that, precisely, as clearly stated in the text of the document and your own comment.

> , it is a serious DOS vulnerability because an account can be created for only 2500 gas (other methods require at least 21000).

that isn't the case, you could only call into an EoA with a 0 nonce with a 7702 tx, and send its balance to a different address. In this case, it would be deleted according to your 'd' rule. You would have to pay the 21000 gas, so no DOS vector.

> Personally I am opposed to a version of EIP-7702 that creates empty accounts. 
the only thing that would prevent that would be to increase the nonce of the target account, which has received a lot of negative feedback since it apparently makes it much more difficult for AA to work properly.

yes, that is what I just heard in ACDE, in which case this EIP is moot. I'm going to move it to withdrawn.

> 
> As it stands, rules a, b and c are uncontroversial.

> Rule d is now a nullity because all empty accounts have been deleted and it is impossible to create new ones.

that is no longer the case once 7702 is rolled out. this is the one that I want to deprecate, in fact.

> 
> (...) it is a serious DOS vulnerability because an account can be created for only 2500 gas (other methods require at least 21000).

this is only possible to trigger if you send a 7702 tx to an empty nonce, and drain its funds. You would have to pay the tx costs and there is no DOS vector.
 
> 
> Personally I am opposed to a version of EIP-7702 that creates empty accounts. If it is decided that EIP-7702 should create empty accounts, then we can then discuss changing the rules about them. Until then this EIP doesn't change the spec in a meaningful way.

The problem is that the only way to fix this would be to increase the nonce of the called contract, which I'm told is wreaking havoc with AA. It's also a problem if an alternate method for the verkle transition is chosen.